### DISCUSSION TITLE
Apple Silicon (M1) support

### DISCUSSION THREAD

**Discussion by muZZkat** [mapsforge/vtm#857] [Ideas]
I was just looking at getting a desktop app working on an Apple Silicon and I need a new libjni for the M1 (arm). I tried to dig into the JNI build scripts, but struggling to work out how to add it to the JNI build scripts.

I added a new target to JniBuilder.java, but that's about all I can work out.

Any hints?

Thanks

**Comment by devemux86** [comment]
VTM works with OpenGL.

Is there OpenGL still on macOS or M1?

**Reply by mcbridet** [reply]
It supports OpenGL 4.1, though it is officially marked as deprecated.

**Reply by devemux86** [reply]
And it has official OpenGL drivers?
Then why does it not work as in the past?

**Reply by muZZkat** [reply]
I've been using it on my Intel Mac for year, the new M1 is arm and needs a new libvtm-jni.dylib (there is only a x86_32 and x86_64 in the package (vim-desktop/natives/osx/)

**Reply by devemux86** [reply]
Based on user interest, any platform other than Android has never been the focus of the library.

And it is impossible to test the library on platforms where developer(s) do not have access.

If you can produce VTM native libs for a new platform, I could include them in repository.

libGDX + LWJGL must support new platform, since vtm-desktop uses libGDX + LWJGL.

**Reply by muZZkat** [reply]
I'm happy to work on it, the build scripts and instructions in the JNI folder don't seem to align with the libvtm-jni binaries in the repo, I couldn't get them to run at all so I was just wondering if anyone knew. I will keep having a play and see if I can get them going.

**Reply by devemux86** [reply]
I created VTM native libs when I revived the library, it was a long time ago (https://github.com/mapsforge/vtm/issues/14).

I used the library scripts and built everything on linux for all platforms.

**Reply by muZZkat** [reply]
I got a build of the libvtm-jni working and now some libgdx issues... I will keep going next week and see if I can get a PR
